system-tests/tests/smoke/cre/workflow_s3_test.go

```diff
@@ -0,0 +1,270 @@
+	// hack to address CRE CLI broken behaviour: `-o` fileName.wasm.br produces `fileName.wasm.br.b64`
+	// and its `upload` command returns:
+	// `Error: failed to create or update object: ... supported extensions: .wasm.br, .json, .yaml, .yml
+	err = os.Rename(wasmFilePath+".b64", wasmFilePath)
+	require.NoError(t, err)
+
+	stats, err := os.Stat(wasmFilePath)
+	require.NoError(t, err)
+	require.NotZero(t, stats.Size())
+
+	provider, err := s3provider.NewMinioFactory().New()
+	require.NoError(t, err)
+
+	mockCreConfig(t, filepath.Dir(workflowFilePath), provider)
+
+	output, err := cli.Upload(
+		crecli.MINIO,
+		wasmFilePath,
+		workflowConfigPath,
+	)
+	require.NoError(t, err)
+
+	fmt.Printf("%#v\n", output)
+
+	require.NoError(t, err, "couldn't load test config")
+	require.Len(t, in.NodeSets, 1, "expected 1 node set in the test config")
+
+	firstBlockchain := in.Blockchains[0]
+
+	chainIDInt, err := strconv.Atoi(firstBlockchain.ChainID)
+	require.NoError(t, err, "failed to convert chain ID to int")
+	chainIDUint64 := libc.MustSafeUint64(int64(chainIDInt))
+
+	mustSetCapabilitiesFn := func(input []*ns.Input) []*keystonetypes.CapabilitiesAwareNodeSet {
+		return []*keystonetypes.CapabilitiesAwareNodeSet{
+			{
+				Input:              input[0],
+				Capabilities:       SinglePoRDonCapabilitiesFlags,
+				DONTypes:           []string{keystonetypes.WorkflowDON, keystonetypes.GatewayDON},
+				BootstrapNodeIndex: 0, // not required, but set to make the configuration explicit
+				GatewayNodeIndex:   0, // not required, but set to make the configuration explicit
+			},
+		}
+	}
+
+	capabilityFactoryFns := []keystonetypes.DONCapabilityWithConfigFactoryFn{
+		computecap.ComputeCapabilityFactoryFn,
+		consensuscap.OCR3CapabilityFactoryFn,
+		croncap.CronCapabilityFactoryFn,
+	}
+
+	universalSetupInput := creenv.SetupInput{
+		CapabilitiesAwareNodeSets:            mustSetCapabilitiesFn(in.NodeSets),
+		CapabilitiesContractFactoryFunctions: capabilityFactoryFns,
+		BlockchainsInput:                     in.Blockchains,
+		JdInput:                              *in.JD,
+		InfraInput:                           *in.Infra,
+		JobSpecFactoryFunctions: []keystonetypes.JobSpecFactoryFn{
+			creconsensus.ConsensusJobSpecFactoryFn(chainIDUint64),
+			cregateway.GatewayJobSpecFactoryFn([]int{}, []string{}, []string{"0.0.0.0/0"}),
+			crecompute.ComputeJobSpecFactoryFn,
+		},
+		ConfigFactoryFunctions: []keystonetypes.ConfigFactoryFn{
+			gatewayconfig.GenerateConfig,
+		},
+	}
+
+	_, err = creenv.SetupTestEnvironment(
+		t.Context(),
+		framework.L,
+		cldlogger.NewSingleFileLogger(t),
+		universalSetupInput,
+	)
+	require.NoError(t, err, "failed to setup test environment")
+
+	creCofigHandle, err := os.Open(path.Join(porRepoPath, CreConfigDefaultFileName))
+	require.NoError(t, err)
+
+	err = crecli.DeployWorkflow(
+		cli.CreCLICommandPath,
+		output.BinaryURL,
+		&output.ConfigURL,
+		nil,
+		creCofigHandle,
+		&porRepoPath,
+	)
+	require.NoError(t, err)
+}
```

Comment:
does it make sense to keep this test? wouldn't it be better to switch other smoke tests to using S3 by default?